File: Agile Development Practices/Code/Java/videoworld/src/main/java/com/thoughtworks/videorental/domain/Customer.java
package com.thoughtworks.videorental.domain;

import java.util.Set;

public class Customer {
	private String name;
	private int frequentRenterPoints = 0;

	public Customer(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public String statement(final Set<Rental> newRentals) {
		String result = "Rental Record for " + getName() + "\n";

		double totalAmount = 0;
		for (Rental rental : newRentals) {
			// show figures for this rental
			final Integer rentalDays = rental.getPeriod().getDuration().getDays();

			result += "  " + rental.getMovie().getTitle() + "  -  $"
					+ String.valueOf(rental.getMovie().getPrice().getCharge(rentalDays)) + "\n";

			totalAmount += rental.getMovie().getPrice().getCharge(rentalDays);

			frequentRenterPoints += rental.getMovie().getPrice().getFrequentRenterPoints(rentalDays);
		}

		// add footer lines
		result += "Amount charged is $" + String.valueOf(totalAmount) + "\n";
		result += "You have a new total of " + String.valueOf(frequentRenterPoints) + " frequent renter points";
		return result;
	}

}

File: Agile Development Practices/Code/DotNet/videoworld/VideoWorld/Models/Customer.cs
using System;
using System.Collections.Generic;
using Ninject;

namespace VideoWorld.Models
{
    public class Customer
    {
        private readonly Cart cart = new Cart();
        private int frequentRenterPoints = 0;

        public Customer(string name)
        {
            Name = name;
        }

        [Inject]
        public Customer()
        {
            Name = "Unknown Customer";
        }

        public Cart Cart
        {
            get { return  cart; }
        }

        protected string Name { get; private set; }

        public string Statement(List<Rental> newRentals)
        {
            String result = "Rental Record for " + Name + "\n";

            decimal totalAmount = 0.00m;
            foreach (Rental rental in newRentals)
            {
                // show figures for this rental
                int rentalDays = rental.Period;

                result += "  " + rental.Movie.Title + "  -  $"
                          + rental.Movie.Price.GetCharge(rentalDays) + "\n";

                totalAmount += rental.Movie.Price.GetCharge(rentalDays);

                frequentRenterPoints += rental.Movie.Price.GetFrequentRenterPoints(rentalDays);
            }

            // add footer lines
            result += "Amount charged is $" + totalAmount + "\n";
            result += "You have a new total of " + frequentRenterPoints + " frequent renter points";
            return result;
        }

    }
}